src/overlays/actors/ovl_En_Sth/z_en_sth.h

```diff
@@ -2,33 +2,49 @@
 #define Z_EN_STH_H
 
 #include "global.h"
+#include "overlays/actors/ovl_En_Si/z_en_si.h"
+#include "objects/object_sth/object_sth.h"
+#include "objects/object_ahg/object_ahg.h" 
+#include "objects/object_mask_truth/object_mask_truth.h"
 
 struct EnSth;
 
 typedef void (*EnSthActionFunc)(struct EnSth*, PlayState*);
 
-#define ENSTH_GET_F(thisx) ((thisx)->params & 0xF)
-#define ENSTH_GET_100(thisx) ((thisx)->params & 0x100)
+typedef enum {
+    /* 1 */ STH_TYPE_UNUSED_1 = 1,
+    /* 2 */ STH_TYPE_SWAMP_SPIDER_HOUSE_CURED, // cursed is EnSsh
+    /* 3 */ STH_TYPE_MOON_LOOKING, // South Clock Town, looking up at the moon
+    /* 4 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_GREET, // looking for shelter
+    /* 5 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_PANIC, // shelter was not enough
+    // Other values: Actor will spawn and animate with arm waving, no further interaction.
+} EnSthTypes;
 
-enum {
-    /* 1 */ ENSTH_F_1 = 1,
-    /* 2 */ ENSTH_F_2,
-    /* 3 */ ENSTH_F_3,
-    /* 4 */ ENSTH_F_4,
-    /* 5 */ ENSTH_F_5,
-};
+// Note: Vanilla types usually have 0xFEXX typing, but this upper section is unused by the code, reason unknown
+#define STH_GET_TYPE(thisx) ((thisx)->params & 0xF)
+#define STH_GET_SWAMP_BODY(thisx) ((thisx)->params & 0x100)
+
+// The get item for the reward for ocean spiderhouse (wallet, or rupees) is set here
+#define STH_GI_ID(thisx) ((thisx)->home.rot.z)
+
+// This actor has its own flags system
+#define STH_FLAG_CLEAR                          (0)
+#define STH_FLAG_DRAW_TRUTH_MASK                (1 << 0)
```

Comment:
Only just now realized this: is there a particular reason you chose `TRUTH_MASK` here instead of `MASK_OF_TRUTH`?

Reply:
only because its shorter, but sure

Reply:
While I see it here, I think we tend to avoid `0` flag defines since it isn't shiftable at all i.e. I don't think `STH_FLAG_CLEAR` is needed. Setting a flag to 0 is pretty clear I think